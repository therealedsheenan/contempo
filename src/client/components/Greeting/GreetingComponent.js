import React from 'react'

export default function GreetingComponent (props) {
  return (
    <div className='Greeting'>Hello Contempo</div>
  )
}
