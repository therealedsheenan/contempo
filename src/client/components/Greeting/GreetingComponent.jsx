import React from 'react';
import PropTypes from 'prop-types';

// Test usage of Async Component
import Async from '../Async/AsyncComponent';

import {
  Title,
  SubTitle,
  Wrapper
} from './styles';


const GreetingComponent = props => (
  <Wrapper>
    <Title>Good {props.message}!</Title>
    <SubTitle>Hello there! Welcome to contempo.</SubTitle>
    <img src="./assets/images/contempo.png" alt="" />
  </Wrapper>
  );

GreetingComponent.defaultProps = {
  message: 'No message passed.'
};

GreetingComponent.propTypes = {
  message: PropTypes.string
};

export default GreetingComponent;
