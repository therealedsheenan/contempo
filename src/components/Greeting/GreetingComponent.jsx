// @flow
import React from 'react';
import { Title, SubTitle, Wrapper } from './styles';

const GreetingComponent = ({ message }: { message?: string }) => (
  <Wrapper>
    <Title>Good {message}!</Title>
    <SubTitle>Welcome to</SubTitle>
    <div>
      <img style={{ width: '100%' }} src="./assets/images/contempo.png" alt="" />
    </div>
  </Wrapper>
);

GreetingComponent.defaultProps = {
  message: 'No message passed'
}

export default GreetingComponent;
