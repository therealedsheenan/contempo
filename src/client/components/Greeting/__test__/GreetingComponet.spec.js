// Link.react-test.js
import React from 'react'
import renderer from 'react-test-renderer'
import GreetingComponent from '../GreetingComponent'

test('Link changes the class when hovered', () => {
  const component = renderer.create(<GreetingComponent />).toJSON()
  expect(component).toMatchSnapshot()
})
