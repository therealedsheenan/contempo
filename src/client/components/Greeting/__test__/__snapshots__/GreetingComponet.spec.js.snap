exports[`test Link changes the class when hovered 1`] = `
<div
  className="Greeting">
  <h1
    className="iEphjE">
    Hello Contempo
  </h1>
</div>
`;
